feat: add tls support

sholdee opened the pull request.

sholdee committed:
feat: add tls support

miaucl commented:
Hey, thanks for the PR. To review this, I will have to setup my own mqtt with tls. This might take a while, especially over the holidays, your PR is not forgotten :)

sholdee committed:
Merge remote-tracking branch 'origin/master' into add-tls

# Conflicts:
#	linux2mqtt/linux2mqtt.py

sholdee committed:
fix: preserve tls config compatibility

sholdee commented:
> Hey, thanks for the PR. To review this, I will have to setup my own mqtt with tls. This might take a while, especially over the holidays, your PR is not forgotten :)

Thanks! Updated PR and fixed merge conflicts. 👍

miaucl reviewed (approved):
Looks good now, still could not test it on my side... too much going on, but trusting you guys that it works.
